lib/invitations/cancelInvitation.ts

```diff
@@ -0,0 +1,52 @@
+import { prisma } from "@lib/integration/prismaConnector";
+import { InvitationNotFoundError } from "./exceptions";
+import { checkPrivileges } from "@lib/privileges";
+import { UserAbility } from "@lib/types";
+
+/**
+ * Cancel an invitation
+ *
+ * @param ability
+ * @param invitationId
+ */
+export const cancelInvitation = async (ability: UserAbility, invitationId: string) => {
+  // Retrieve the invitation
+  const invitation = await prisma.invitation.findUnique({
+    where: {
+      id: invitationId,
+    },
+    select: {
+      email: true,
+      templateId: true,
+    },
+  });
+
+  if (!invitation) {
+    throw new InvitationNotFoundError();
+  }
+
+  checkPrivileges(ability, [
+    { action: "update", subject: { type: "FormRecord", object: { id: invitation.templateId } } },
+  ]);
+
+  // If no invitation found, return an error
+  if (!invitation) {
```

Comment:
I believe this might be a duplicate of line 24

Reply:
Indeed (removed)